src/core/util/mod.rs

```diff
@@ -15,57 +14,66 @@ pub(crate) fn base_request() -> http::request::Builder {
     Request::builder().header("Prefer", "OID4VP-0.0.20")
 }
 
-#[derive(Debug, Clone)]
-pub struct ReqwestClient(reqwest::Client);
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
Does it mean that the "client" code isn't fully available on WASM? Why not use reqwest's default WASM client (https://docs.rs/reqwest/latest/reqwest/#wasm) and simply disable rusttls?